mysql/infoschema.go

```diff
@@ -304,6 +311,74 @@ func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]str
 	return primaryKeys, m, nil
 }
 
+type fkConstraint struct {
+	name    string
+	table   string
+	refcols []string
+	cols    []string
+}
+
+// getForeignKeys return list all the foreign keys constraints.
+// MySQL supports cross-database foreign key constraints. We ignore
+// them because HarbourBridge works database at a time (a specific run
+// of HarbourBridge focuses on a specific database) and so we can't handle
+// them effectively.
+func getForeignKeys(conv *internal.Conv, db *sql.DB, table schemaAndName) (foreignKeys []schema.ForeignKey, err error) {
+	q := `SELECT k.REFERENCED_TABLE_NAME,k.COLUMN_NAME,k.REFERENCED_COLUMN_NAME,k.CONSTRAINT_NAME
+		FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS AS t 
+		INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS k 
+			ON t.CONSTRAINT_NAME = k.CONSTRAINT_NAME 
+			AND t.CONSTRAINT_SCHEMA = k.CONSTRAINT_SCHEMA 
+			AND t.TABLE_NAME = k.TABLE_NAME 
+			AND k.REFERENCED_TABLE_SCHEMA = k.TABLE_SCHEMA
+		WHERE k.TABLE_SCHEMA = ? 
+			AND k.TABLE_NAME = ? 
+			AND t.CONSTRAINT_TYPE = "FOREIGN KEY" 
+		ORDER BY
+			k.REFERENCED_TABLE_NAME,
+			k.COLUMN_NAME,
+			k.ORDINAL_POSITION;`
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var col, refCol, refTable, fKeyName string
+	fKeys := make(map[string]fkConstraint)
+	var keyNames []string
+
+	for rows.Next() {
+		err := rows.Scan(&refTable, &col, &refCol, &fKeyName)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
+			continue
+		}
+		if _, found := fKeys[fKeyName]; found {
+			fk := fKeys[fKeyName]
+			fk.cols = append(fk.cols, col)
+			fk.refcols = append(fk.refcols, refCol)
+			fKeys[fKeyName] = fk
+			continue
+		}
+		var mCols []string
```

Comment:
Suggest using the more idiomatic (and shorter) sequence:

mCols := []string{col}
mRefCols := []string{refCol}

and I'd probably go a step further, and just inline these and get rid of mCols and mRefCols completely i.e.

fKeys[fKeyName] = fkConstraint{name: fKeyName, table: refTable, refcols: []string{refCol}, cols: []string{col}}

Reply:
thanks nevin, updated this